p2p/crypto/secio/rw.go

```diff
@@ -105,66 +101,27 @@ func (r *etmReader) NextMsgLen() (int, error) {
 	return r.msg.NextMsgLen()
 }
 
-func (r *etmReader) drainBuf(buf []byte) int {
-	if r.buf == nil {
-		return 0
-	}
-
-	n := copy(buf, r.buf)
-	r.buf = r.buf[n:]
-	return n
-}
-
 func (r *etmReader) Read(buf []byte) (int, error) {
 	r.Lock()
 	defer r.Unlock()
 
-	// first, check if we have anything in the buffer
-	copied := r.drainBuf(buf)
-	buf = buf[copied:]
```

Comment:
I think this was meant to deal with the case where `buf` is smaller than `r.buf` but I think it should read `buf = buf[:copied]` (all the bytes up to `copied`, not all the bytes after).